include/seqan3/io/sequence/sequence_file_format_fasta.hpp

```diff
@@ -94,6 +98,122 @@ namespace seqan3
+    void read(stream_type        & stream,
+              options_type const & options,
+              seq_type           & sequence,
+              id_type            & id,
+              qual_type          & SEQAN3_DOXYGEN_ONLY(qualities),
+              seq_qual_type      & seq_qual)
+    {
+        static_assert(detail::decays_to_ignore_v<seq_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the sequence field, or the seq_qual field need to be set to std::ignore.");
+        static_assert(detail::decays_to_ignore_v<qual_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the qualities field, or the seq_qual field need to be set to std::ignore.");
+
+        auto stream_view = ranges::iterator_range{std::istreambuf_iterator<char>{stream},
+                                                  std::istreambuf_iterator<char>{}};
+        // ID
+        read_id(stream_view, options, id);
+
+        // Sequence
+        if constexpr (!detail::decays_to_ignore_v<seq_type>) // sequence
+            read_seq(stream_view, options, sequence);
+        else
+            read_seq(stream_view, options, seq_qual);        // seq_qual (possibly std::ignore, too)
+
+        // make sure "buffer at end" implies "stream at end"
+        if ((std::istreambuf_iterator<char>{stream} == std::istreambuf_iterator<char>{}) &&
+            (!stream.eof()))
+        {
+            stream.get(); // triggers error in stream and sets eof
+        }
+    }
+
+    //!\copydoc sequence_file_out_format_concept::write
+    template <typename stream_type,     // constraints checked by file
+              typename options_type,    // given by file
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type,
+              typename seq_qual_type>
+    void write(stream_type         & stream,
+               options_type  const & options,
+               seq_type           && sequence,
+               id_type            && id,
+               qual_type          && SEQAN3_DOXYGEN_ONLY(qualities),
+               seq_qual_type      && seq_qual)
+    {
+        static_assert(detail::decays_to_ignore_v<seq_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the sequence field, or the seq_qual field need to be set to std::ignore.");
+        static_assert(detail::decays_to_ignore_v<qual_type> || detail::decays_to_ignore_v<seq_qual_type>,
+                      "Either the qualities field, or the seq_qual field need to be set to std::ignore.");
+
+        ranges::ostreambuf_iterator stream_it{stream};
+
+        // ID
+        if constexpr (detail::decays_to_ignore_v<id_type>)
+        {
+            throw std::logic_error{"The ID field may not be set to ignore when writing FASTA files."};
+        }
+        else
+        {
+            if (ranges::empty(id)) //[[unlikely]]
```

Comment:
Do these [[unlikely]] tags really have to stay in comment? As far as I understood, we want to include them, because they start working automatically.

Reply:
yeah, I wish, see #275